pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -466,6 +467,8 @@ private static double[] toDoubleArray(Object value) {
           doubleValues[i] = longValues[i];
         }
         return doubleValues;
+      } else if (value instanceof DoubleArrayList) {
```

Comment:
(minor) Suggest moving this between `double[]` and `int[]` case because this branch could be hit in normal case, other branches will only be hit when different segments have different schema

Reply:
moved